src/lib/common.c

```diff
@@ -404,6 +404,61 @@ char *replace_cntrl_chars(const char *str, char c) {
 	return rv;
 }
 
+char *escape_cntrl_chars(const char *str) {
+	if (str) {
+		unsigned int cntrl_chars = 0;
+		const char *c = str;
+		while (*c) {
+			switch (*c++) {
+				case '\b':
+				case '\a':
+				case '\e':
+				case '\f':
+				case '\n':
+				case '\r':
+				case '\t':
+				case '\v':
+				case '\"':
+				case '\'':
+				case '\?':
+				case '\\':
+					++cntrl_chars;
+				default: break;
+			}
+		}
+		char *rv = malloc(strlen(str) + cntrl_chars + 1);
+		char *ptr = rv;
+		if (!rv)
+			errExit("malloc");
+		c = str;
+		while (*c) {
+			if (iscntrl(*c)) {
+				*ptr++ = '\\';
+				switch (*c) {
+					case '\b': *ptr++ = 'b'; break;
+					case '\a': *ptr++ = 'a'; break;
+					case '\e': *ptr++ = 'e'; break;
+					case '\f': *ptr++ = 'f'; break;
+					case '\n': *ptr++ = 'n'; break;
+					case '\r': *ptr++ = 'r'; break;
+					case '\t': *ptr++ = 't'; break;
+					case '\v': *ptr++ = 'v'; break;
+					case '\"': *ptr++ = '\"'; break;
+					case '\'': *ptr++ = '\''; break;
+					case '\?': *ptr++ = '?'; break;
+					case '\\': *ptr++ = '\\'; break;
+				}
+			} else {
+				*ptr++ = *c;
+			}
+			c++;
+		}
+		*ptr = '\0';
+		return rv;
+	}
+	return NULL;
+}
+
```

Comment:
```suggestion
char *escape_cntrl_chars(const char *str) {
	if (str == NULL)
		return NULL;

	size_t cntrl_chars = 0;
	const char *c = str;
	while (*c) {
		switch (*c++) {
		case '\b':
		case '\a':
		case '\e':
		case '\f':
		case '\n':
		case '\r':
		case '\t':
		case '\v':
		case '\"':
		case '\'':
		case '\?':
		case '\\':
			++cntrl_chars;
		default:
			break;
		}
	}

	char *rv = malloc(strlen(str) + cntrl_chars + 1);
	if (!rv)
		errExit("malloc");

	char *ptr = rv;
	c = str;
	while (*c) {
		if (iscntrl(*c)) {
			*ptr++ = '\\';
			switch (*c) {
			case '\b': *ptr++ = 'b'; break;
			case '\a': *ptr++ = 'a'; break;
			case '\e': *ptr++ = 'e'; break;
			case '\f': *ptr++ = 'f'; break;
			case '\n': *ptr++ = 'n'; break;
			case '\r': *ptr++ = 'r'; break;
			case '\t': *ptr++ = 't'; break;
			case '\v': *ptr++ = 'v'; break;
			case '\"': *ptr++ = '\"'; break;
			case '\'': *ptr++ = '\''; break;
			case '\?': *ptr++ = '?'; break;
			case '\\': *ptr++ = '\\'; break;
			}
		} else {
			*ptr++ = *c;
		}
		c++;
	}

	*ptr = '\0';
	return rv;
}

```

Style:

* Try to return early to reduce indentation
* Do not indent `case` statements
* Use `size_t` instead of `unsigned int` when counting to make it more obvious
  (the compiler should hopefully make them equivalent anyway)
 
Also, I was a bit confused when reading the rv/ptr declarations; I think that
moving `if (!rv)` right below `char *rv =` makes it a bit clearer (even if
it's maybe less pretty).